src/stores/globalStore.js

```diff
@@ -53,5 +53,31 @@ export const globalStore = createStore(
       userStorage.reset();
       return { ...state, currentUser: null, loggedIn: false };
     },
+    addPost(state, { content }) {
+      const post = {
+        id: state.posts.length + 1,
+        author: state.currentUser.username,
+        time: Date.now() - 0 * 분,
+        content,
+        likeUsers: [],
+      };
+      return { ...state, posts: [...state.posts, post] };
+    },
+    likePost(state, postId) {
+      console.log(postId);
+      const { username } = state.currentUser;
+      const postIndex = state.posts.findIndex((post) => post.id === postId);
+      const likeUsers = state.posts[postIndex].likeUsers;
+
+      if (state.posts[postIndex].likeUsers.includes(username)) {
+        const userIndex = likeUsers.findIndex((user) => user === username);
+        likeUsers.splice(userIndex, 1);
+      } else {
+        likeUsers.push(username);
+      }
```

Comment:
위의 부분에서 원본 배열을 직접 변경해주시고 있는 구조 같은데, 저는 원본 객체를 변경하면 이전 상태와 현재 상태를 구별하기가 힘들더라고요 ㅎㅎ

```suggestion
if (post.likeUsers.includes(username)) {
  newLikeUsers = post.likeUsers.filter(user => user !== username);
} else {
  newLikeUsers = [...post.likeUsers, username];
}
```

splice(), push() 대신 filter(), ...를 써서 새로운 배열을 반환하여 불변성을 유지하는 구조는 어떨까 한번 생각해 봤습니다!